zerver/actions/users.py

```diff
@@ -163,7 +163,7 @@ def do_deactivate_user(
         if settings.BILLING_ENABLED:
             update_license_ledger_if_needed(user_profile.realm, event_time)
 
-    delete_user_sessions(user_profile)
+    transaction.on_commit(lambda: delete_user_sessions(user_profile))
```

Comment:
Should this be inside the `transaction.atomic` block just above? That seems more obviously correct to me. 

(We should also probably move the `send_event` calls below to be `transaction.on_commit()` triggered as well, though that can be a second commit).

Reply:
> Should this be inside the transaction.atomic block just above? That seems more obviously correct to me.

I think they're functionally pretty much the same in this case, but yeah putting it in the `atomic` block is more obvious and cleaner, fixed.

Added a commit for send_event as well